### ISSUE TITLE
useIntervalEffect

### ISSUE THREAD

**Issue by ctrlplusb** [streamich/react-use#228]
Would this be considered?

**Source**

```typescript
import { useEffect } from "react";

export default function useIntervalEffect(
  effect: Function,
  ms: number,
  dependencies: any[] = []
) {
  useEffect(() => {
    const interval = setInterval(effect, ms);
    return () => clearInterval(interval);
  }, dependencies);
}
```

**Example 1**

```typescript
import useIntervalEffect from "./useIntervalEffect";

function App() {
  useIntervalEffect(() => {
    console.log("Effect run");
  }, 2000);

  return <div>Hello world</div>;
}
```

**Example 2: With dependencies**

When the effect depends on an external value, the value(s) needs to be provided as an argument, similar to the standard `useEffect`. This starts a new interval with each dependency change, which should be fine in most cases, as shown below.

```typescript
import useIntervalEffect from "./useIntervalEffect";

function Counter() {
  const [counter, setCounter] = useState(0);
  useIntervalEffect(() => setCounter(counter + 1), 1000, [counter]);
  return <div>{counter}</div>;
}
```

**Demo** 

You can see a demo of both versions here: https://codesandbox.io/s/131wwmopr7

**Comment by jiangbo2015** [comment]
named `useInterval` is better

**Comment by ctrlplusb** [comment]
@streamich your call? Should I make a PR and if so should we name `useIntervalEffect` or `useInterval`. Personally I prefer the `useIntervalEffect` as it is more explicit that it is an interval bound effect.

**Comment by streamich** [comment]
@ctrlplusb Yes, the hook would be useful, naming is up to you.

**Comment by brandonkal** [comment]
Rather than providing dependencies for the function, a cleaner approach would be to wrap the function. This way, the function can be updated without closure concern. Checking for null, should also be done.

```js
export function useInterval(callback: Function, delay: number | null) {
  const savedCallback = React.useRef<Function>()

  // Remember the latest callback.
  React.useEffect(() => {
    savedCallback.current = callback
  }, [callback])

  // Set up the interval.
  React.useEffect(() => {
    function tick() {
      savedCallback.current!()
    }
    if (delay !== null) {
      let id = setInterval(tick, delay)
      return () => clearInterval(id)
    }
  }, [delay])
}
```
More details: https://overreacted.io/making-setinterval-declarative-with-react-hooks/

**Comment by ctrlplusb** [comment]
Nice! Happy for you to take over on this @brandonkal 👍

**Comment by wardoost** [comment]
Was implemented in #470
